terraform/cross-device/README.md

```diff
@@ -0,0 +1,53 @@
+# Cross-device Federated Learning
+
+This module is an example of an end to end demo for cross-device Federated Learning. This example deploys 6 different workloads:
+- `aggregator`: this is an offline job that reads device gradients and calculates aggregated result with DP
+- `collector`: this is an offline job that runs periodically to query active task and encrypted gradients, resulting in deciding when to kick off aggregating
+- `modelupdater`: this is an offline job that listens to events and publishes results so that device can download
+- `task-assignment`: this is a front end service that distributes training tasks to devices
+- `task-management`: this is an offline job that manages tasks
+- `task-scheduler`: this is an offline job that either runs periodically or is triggered by some events
+
+This example builds on top of the infrastructure that the
+[blueprint provides](../../../../README.md), and follows the best practices the
+blueprint establishes.
+
+## Prerequisites
+
+- A POSIX-compliant shell
+- Git (tested with version 2.41)
+- Docker (tested with version 20.10.21)
+
+## Infrastructure
+
+It creates:
+- A spanner instance for storing the status of training
+- Pubsub topics that act as buses for messages between microservices
+- Buckets for storing the trained models
+
+To deploy this solution, just set the `cross-device` flag to `true`.
+
+To ensure end-to-end confidentiality, you need to enable confidential nodes.
+
+However, it is also necessary to use VM families that support this feature, such as **N2D** or **C2D**.
+When using confidential nodes, set `enable_confidential_nodes` to `true` and `cluster_tenant_pool_machine_type` to `n2d-standard-8`. In addition, in order to have the minimum number of replicas required during deployment, you need at least 4 nodes.
```

Comment:
Remove this because it's duplicating what we wrote in the variable definition

Reply:
And move the bits about the machine type in the related variable description (see my other comment).